analyses/tmb-vaf-longitudinal/01-preprocess-data.Rmd

```diff
@@ -0,0 +1,175 @@
+)
+
+# Read in maf file, calculate VAF, and retain only coding gene mutations
+# We will use this column to label gene mutations in the final graphs
+maf <- data.table::fread(maf_file, data.table = FALSE) %>% 
+  filter(Tumor_Sample_Barcode %in% pbta$Kids_First_Biospecimen_ID,
+         Variant_Classification %in% maf_nonsynonymous) %>%
+  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% # change name of the biospecimen to match the one from the histologies files
+  dplyr::mutate(VAF = t_alt_count / (t_ref_count + t_alt_count), # calculate VAF 
+                gene_protein = paste(Hugo_Symbol, HGVSp_Short, sep = "_")) %>%  # concatenate gene and protein information
+  select(Kids_First_Biospecimen_ID, Hugo_Symbol, Chromosome, Start_Position, Variant_Classification, VAF, HGVSc, HGVSp, HGVSp_Short, gene_protein) 
+
+# Read in matched_time_points_file and get the list of patients 
+# with matched time points for the genomic assays
+patient_list <- readr::read_tsv(matched_time_points_file, guess_max = 100000, show_col_types = FALSE) %>%
+  subset(select = c("Kids_First_Participant_ID", "descriptors"))
+
+# How many patients have paired time samples?
+paired_time_samples <- length(unique(patient_list$Kids_First_Participant_ID))
+
+# Add metadata from pbta to patient list
+patient_list_metadata <- patient_list %>% 
+  left_join(pbta, by = "Kids_First_Participant_ID") %>%
+  filter(!(experimental_strategy == "RNA-Seq")) %>%
+  mutate(tumor_descriptor = case_when(grepl("Primary Tumor", tumor_descriptor) ~ "Diagnosis", 
+                                      grepl("Initial CNS Tumor", tumor_descriptor) ~ "Diagnosis", 
+                                      grepl("Progressive Disease Post-Mortem", tumor_descriptor) ~ "Deceased", 
+                                      TRUE ~ tumor_descriptor)) 
+```
+
+# Create maf file for autopsy samples
+```{r maf_autopsy}
+# We will filter the df to keep paired smaples with Deceased timepoint (autopsy samples).
+# There are patient cases with more than one time points in addition to Deceased, e.g. Diagnosis and Progressive.
+# There are also multiple biospecimen samples per timepoint and patient case.
+# We will create unique identifiers for each case that will enable to create one plot for each case.
+maf_autopsy <- patient_list_metadata %>%
+  mutate(Deceased = case_when(grepl("Deceased", descriptors) ~ "Deceased")) %>%
+  filter(Deceased == "Deceased") %>% 
+  left_join(maf, by = "Kids_First_Biospecimen_ID") %>%
+  filter(!is.na(Kids_First_Participant_ID),
+         !is.na(VAF),
+         !VAF == 0, 
+         !gene_protein == "Unknown") %>%
+  select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, descriptors, tumor_descriptor, experimental_strategy, Hugo_Symbol, gene_protein, VAF) %>%
+  mutate(match_id = paste(Kids_First_Biospecimen_ID, tumor_descriptor, gene_protein, VAF, sep = "_"),
+         match_id_bs = paste(tumor_descriptor, Kids_First_Biospecimen_ID, sep = "_"),
```

Comment:
I'm unsure what the purpose of these columns is, but maybe it will be clearer when reviewing subsequent scripts.